src/text/rich_text.rs

```diff
@@ -23,6 +25,14 @@ pub struct RichText {
     links: Arc<[Link]>,
 }
 
+impl PartialEq for RichText {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.buffer, &other.buffer) && Arc::ptr_eq(&self.attrs, &other.attrs) && Arc::ptr_eq(&self.links, &other.links)
+    }
+}
+
+impl Eq for RichText {}
```

Comment:
This is an invalid implementation of `Eq` and a dubious implementation of `PartialEq`. Two buffers with different pointer values but identical content shouldn't return false in PartialEq.

As a placeholder solution, I'd suggest adding a function called eg `maybe_eq` to the TextStorage trait with the same semantic as the now-removed `Data::same()`. We can think of a better design once this is merged.